python/cugraph/cugraph/link_prediction/jaccard.py

```diff
@@ -72,21 +99,28 @@ def jaccard(input_graph, vertex_pair=None, do_expensive_check=True):
     ----------
     input_graph : cugraph.Graph
         cuGraph Graph instance, should contain the connectivity information
-        as an edge list (edge weights are not used for this algorithm). The
+        as an edge list (edge weights are not supported yet for this algorithm). The
         graph should be undirected where an undirected edge is represented by a
         directed edge in both direction. The adjacency list will be computed if
         not already present.
 
+        This implementation only supports undirected, unweighted Graph.
+
     vertex_pair : cudf.DataFrame, optional (default=None)
         A GPU dataframe consisting of two columns representing pairs of
         vertices. If provided, the jaccard coefficient is computed for the
         given vertex pairs.  If the vertex_pair is not provided then the
         current implementation computes the jaccard coefficient for all
         adjacent vertices in the graph.
 
-    do_expensive_check: bool (default=True)
-        When set to True, check if the vertices in the graph are (re)numbered
-        from 0 to V-1 where V is the total number of vertices.
+    do_expensive_check : bool, optional (default=False)
+        Deprecated, no longer needed
```

Comment:
Can we provide a better explanation (ie. why this is no longer needed)?  Deprecation messages should also include the alternative the user should use, if possible, but in this case it sounds like there is no alternative?  Does this mean users who used this for input validation now have no alternative, or that all checks are "expensive" now?

I see the PLC API still has `do_expensive_check`, so is there a reason we're removing it here?

Also, for all deprecations, there's some guidelines we're starting to more strictly follow [here](https://docs.rapids.ai/api/cudf/nightly/developer_guide/contributing_guide/#deprecating-and-removing-code).

Reply:
> but in this case it sounds like there is no alternative

Yes there is no alternative for. This expensive check was put in place to check that the vertices were numbered with consecutive integers starting from zero. In fact, failure to do that and making a prior call like `G.number_of_vertices` would yield incorrect results for `Jaccard`. A description must be added explicitly stating that the vertices no longer need to be numbered.